zerver/views/messages.py

```diff
@@ -264,15 +264,20 @@ def by_pm_with(self, query, operand, maybe_negate):
 
     def by_search(self, query, operand, maybe_negate):
         # type: (Query, str, ConditionTransform) -> Query
+        rendered_operand = render_markdown(None, operand)
+        for remove in ('<p>', '</p>', '<br>'):
+            rendered_operand = rendered_operand.replace(remove, '')
```

Comment:
The idea is to remove some very common tags, so we don't accidentally match unrelated messages. I think we won't be able to add tests to it though, since if it isn't there it would probably fail all tests anyway :smiley:.